pkg/target/vapp_dto.go

```diff
@@ -48,61 +78,36 @@ func (vapp *VirtualApp) getCommoditiesBought(vAppBuilder *builder.EntityDTOBuild
 			}
 
 			app := container.App
-			appCommodity, err := builder.NewCommodityDTOBuilder(proto.CommodityDTO_TRANSACTION).
-				Key(app.UUID).
-				Used(app.QPS.Used).
-				Create()
+			appCommodity, err := vapp.createAppCommodity(pod, container, proto.CommodityDTO_TRANSACTION,
+				app.QPS.Used, app.QPS.Capacity)
 			if err != nil {
-				glog.Errorf("failed to create commodity bought for VirtualApp[%s]-pod[%s]-container[%s]-app[%s]",
-					vapp.Name, pod.Name, container.Name, app.Name)
 				continue
 			}
-
+			qpsused += app.QPS.Used
+			qpscap += app.QPS.Capacity
 			bought := []*proto.CommodityDTO{appCommodity}
 
+			if app.ResponseTime != nil {
+				appCommodity, err := vapp.createAppCommodity(pod, container, proto.CommodityDTO_RESPONSE_TIME,
+					app.ResponseTime.Used, app.ResponseTime.Capacity)
+				if err == nil {
+					rtused += app.ResponseTime.Used
+					rtcap += app.ResponseTime.Capacity
+					bought = append(bought, appCommodity)
+				}
+			}
+
 			appProvider := builder.CreateProvider(proto.EntityDTO_APPLICATION, app.UUID)
 			vAppBuilder.Provider(appProvider).BuysCommodities(bought)
 			i++
 		}
 	}
 
 	if i < 1 {
-		return fmt.Errorf("cannot get commodities bought from containers for VApp[%s/%s]", vapp.Kind, vapp.Name)
+		return fmt.Errorf("cannot get commodities bought from containers for VApp[%s/%s]", vapp.Kind, vapp.Name),
+			0.0, 0.0, 0.0, 0.0
 	}
 
-	return nil
+	fi := float64(i)
+	return nil, qpsused / fi, qpscap / fi, rtused / fi, rtcap / fi
```

Comment:
The overall `QPS` should be the **SUM** of all the sub `QPS`;
The overall `ResponseTime` should be the ** (weighted) Average** of the all the sub `ResponseTime`.